test/compressionworker.ts

```diff
@@ -0,0 +1,222 @@
+      assert.equal(payloads[1].events.length, 1);
+      assert.equal(payloads[1].events[0].type, secondMockEventName);
+      done();
+    }
+  });
+
+  it("validates that single event is sent when its own length is above the limit", (done: DoneFn) => {
+    let worker = createTestWorker();
+    let mockEvent = getMockEvent();
+    let mockEventData = Array(Math.round(config.batchLimit + 1)).join("1");
+    mockEvent.state = { data: mockEventData };
+    let addEventMessage = createAddEventMessage(mockEvent);
+
+    messageHandlers.push(messageHandler);
+    worker.postMessage(JSON.stringify(addEventMessage));
+    scheduleTestFailureTimeout();
+
+    function messageHandler(message: IWorkerMessage) {
+      assert.equal(message.type, WorkerMessageType.Upload);
+      let uploadMessage = message as IUploadMessage;
+      let payload = JSON.parse(uploadMessage.rawData);
+      assert.equal(payload.events.length, 1);
+      assert.equal(payload.events[0].type, MockEventName);
+      done();
+    }
+  });
+
+  it("validates that queued payload is sent on terminate", (done: DoneFn) => {
+    let worker = createTestWorker();
+    let mockEvent = getMockEvent();
+    let addEventMessage = createAddEventMessage(mockEvent);
+    let terminateMsg: ITimestampedWorkerMessage = {
+      type: WorkerMessageType.Terminate,
+      time: new Date().getTime()
+    };
+
+    assert.equal(workerMessages.length, 0);
+    messageHandlers.push(messageHandler);
+    worker.postMessage(JSON.stringify(addEventMessage));
+    worker.postMessage(JSON.stringify(terminateMsg));
+    scheduleTestFailureTimeout();
+
+    function messageHandler(message: IWorkerMessage) {
+      assert.equal(message.type, WorkerMessageType.Upload);
+      let uploadMessage = message as IUploadMessage;
+      let payload = JSON.parse(uploadMessage.rawData);
+      assert.equal(payload.events.length, 1);
+      assert.equal(payload.events[0].type, MockEventName);
+      done();
+    }
+  });
+
+  it("validates that payloads consisting of a single XHR error event are not uploaded", (done: DoneFn) => {
+    let worker = createTestWorker();
+    let mockXhrErrorEvent = getMockEvent(InstrumentationEventName);
+    let mockXhrErrorEventState = {
+      type: Instrumentation.XhrError
+    } as IXhrErrorEventState;
+    mockXhrErrorEvent.state = mockXhrErrorEventState;
+    let addEventMessage: IAddEventMessage = {
+      type: WorkerMessageType.AddEvent,
+      time: new Date().getTime(),
+      event: mockXhrErrorEvent
+    };
+    let forceUploadMessage = createForceUploadMessage();
+
+    messageHandlers.push(messageHandler);
+    worker.postMessage(JSON.stringify(addEventMessage));
+    worker.postMessage(JSON.stringify(forceUploadMessage));
+    scheduleTestSuccessTimeout(done);
+
+    function messageHandler(message: IWorkerMessage) {
+      // This handler should NOT be invoked, so intentional assertion failure in this path
+      assert.equal(true, false);
```

Comment:
Isn't there a simple "fail" for this case?

Reply:
Yeah, turns out async 'done' object has fail method. Thanks !

---
In reply to: [136876665](https://github.com/Microsoft/clarity-js/pull/28#discussion_r136876665) [](ancestors = 136876665)